packages/ember-data/tests/integration/serializers/embedded_records_mixin_test.js

```diff
@@ -998,6 +1003,48 @@ test("extractSingle with polymorphic hasMany", function() {
 
 });
 
+test("extractSingle with polymorphic belongsTo", function() {
+  expect(2);
+
+  SuperVillain.reopen({
+    secretLab: DS.belongsTo("secretLab", {polymorphic: true}),
+  });
+
+  env.container.register('adapter:superVillain', DS.ActiveModelAdapter);
+  env.container.register('serializer:superVillain', DS.ActiveModelSerializer.extend(DS.EmbeddedRecordsMixin, {
+    attrs: {
+      secretLab: {embedded: 'always'}
+    }
+  }));
+  var serializer = env.container.lookup("serializer:superVillain");
+
+  var json_hash = {
+    super_villain: {
+      id: "1",
+      first_name: "Tom",
+      last_name: "Dale",
+      secret_lab: {
+        id: "1",
+        type: "BatCave",
+        infiltrated: false
+      }
+    }
+  };
+
+  var json = serializer.extractSingle(env.store, SuperVillain, json_hash);
+
+  deepEqual(json, {
+    id: "1",
+    firstName: "Tom",
+    lastName: "Dale",
+    secretLab: "1",
+    secretLabType: "batCave"
```

Comment:
Our current internal expectation for belongsTo is `{ secretLab: "1", secretLabType: "batCave" }` but we might want to change that in the future to be consistent with the hasMany, cc @cyril-sf